src/devices/machine/6522via.cpp

```diff
@@ -463,7 +463,7 @@ void via6522_device::device_timer(emu_timer &timer, device_timer_id id, int para
 			{
 				if (SI_O2_CONTROL(m_acr) || SO_O2_CONTROL(m_acr))
 				{
-					m_shift_timer->adjust(clocks_to_attotime(1) / 2);
+					m_shift_timer->adjust(clocks_to_attotime(1));
```

Comment:
@cuavas has a complete rewrite of the via6522 on a branch which hopefully will correct this and other problems (most of the Macs don't work because of it) but in the meantime I think this probably makes sense to include.

Reply:
Also, you are talking about CB1, which is the shift clock, but as far as I can see on this schematic CB1 of the VIA6522 is not connected anywhere: [https://console5.com/techwiki/images/c/cb/Vectrex---Logic-Board.png](url) CB2 output the data However, the CB2 controls the *BLANK signal which is used to generate the raster effect for characters so that is why the change of the shift rate affect the text width. I think there is  a bug in the Vectrex driver, I'll have a look.

Reply:
As I read the datasheet, bits are shifted out at half the system clock rate. If we agree on that then you have to invert CB1 every clock cycle which is what the patch does.

Reply:
I get it, CB1 was not part of the fix I did for flanks required for the prodigy driver, I missed that. Will issue PR for that once I understand the implications for the Vectrex driver.

So with my previous fix the shift timer is called once for each flank (twice per clock cycle) in order to get 50/50 duty cycle and to be able to trigger both on raising and falling flanks with the correct timing. The CB1 should toggle and the handler should be called only on raising O2 system clock flanks resulting in a division with 2. 

However, I can't see that vectrex driver uses CB1. The shift_out() function shifts out only on raising CB1 flanks already but CB1 does on every flank. So this probably explain some regression we have seen but not why the Vectrex driver is fixed for those games with this PR.

Reply:
Sure, it wasn't an easy fix admittedly and the original driver didn't give much room for flanks as they were both triggered at the same point of each clock cycle. Still having drivers depend on faulty timing is not great either. If you have a new refactored device I'd be happy to test it out.